lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
         self.confirmation_token = nil
         generate_remember_token
+        save(validate: false)
```

Comment:
They all need to be updated. To complete the password rotation. I can't
split it into 2 calls to save (one with validate false) in a transaction
because the second save would be subject to the same validation issues we
have now.

Best I can do is to set them before calling valid and checking for errors
on those fields too.

On Friday, January 16, 2015, Joe Ferris <notifications@github.com> wrote:

> In lib/clearance/user.rb
> <https://github.com/thoughtbot/clearance/pull/521#discussion-diff-23114734>
> :
>
> >          self.confirmation_token = nil
> >          generate_remember_token
> > +        save(validate: false)
>
> Should you make sure to only update the relevant password attributes here,
> since we're not sure the others are valid?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thoughtbot/clearance/pull/521/files#r23114734>.
>